src/geotop/energy.balance.cc

```diff
@@ -776,18 +776,18 @@ short PointEnergyBalance(long i, long r, long c, double Dt, double JDb,
           else
             {
               k = k_thermal(  0, 1,
-                              Arithmetic_Mean(A->E->Dlayer->co[ns], A->E->Dlayer->co[ns+1],
-                                              A->E->liq->co[ns]/(rho_w*A->E->Dlayer->co[ns]),
-                                              A->E->liq->co[ns+1]/(rho_w*A->E->Dlayer->co[ns+1])),
-                              Arithmetic_Mean(A->E->Dlayer->co[ns], A->E->Dlayer->co[ns+1],
-                                              A->E->ice->co[ns]/(rho_w*A->E->Dlayer->co[ns]),
-                                              A->E->ice->co[ns+1]/(rho_w*A->E->Dlayer->co[ns+1])),
-                              Arithmetic_Mean(A->E->Dlayer->co[ns], A->E->Dlayer->co[ns+1], 1.,
+                              Arithmetic_Mean((*A->E->Dlayer)(ns), (*A->E->Dlayer)(ns+1),
+                                              (*A->E->liq)(ns)/(rho_w*(*A->E->Dlayer)(ns)),
+                                              (*A->E->liq)(ns+1)/(rho_w*(*A->E->Dlayer)(ns+1))),
+                              Arithmetic_Mean((*A->E->Dlayer)(ns), (*A->E->Dlayer)(ns+1),
+                                              A->E->ice->co[ns]/(rho_w*(*A->E->Dlayer)(ns)),
```

Comment:
why not `ice->co`?

Reply:
I'm sobstituting an object at a time, to avoid too many commits. 
And yes, all 1D tests are passing; for 3D, there are run time exceptions that I'm trying to solve.

Reply:
so is `A->E->ice` different from `egy->ice` used in other parts of the code? because there you are using `(*egy->ice)()`

Reply:
In this pull request I was just looking ad **Dlayer**;.
As far as I understand ```A->E->ice``` is the same as ```egy->ice``` (thus ```A->E->Dlayer``` is the same as ```egy->Dlayer```).
The problem was not accessing an element, doing something like (*egy->Dlayer)(), but in the specific function ```calc_C```, in which I preferred to pass a reference instead of a pointer. In any case I want to be able to access ```D(l)``` in the function (possibly using ```()```) so I chose this way.
Hope it's clear, but if not tell me!

Reply:
so if they are the same, you should use the `operator()` also on that object in that line